src/editor/src/routes.js

```diff
@@ -0,0 +1,49 @@
+// Returns the URL's paths
+function splitURL(urlPath) {
+    return urlPath.substr(6).split('/');
+}
+
+// Determines whether path is a file
+function isFile(pathname) {
+    return pathname.split('/').pop().indexOf('.') > -1;
+}
+
+// Determines whether path is a directory
+function isDir(pathname) { return !isFile(pathname); }
```

Comment:
All this `isDir` and `isFile` stuff should happen using `fs.stat` since you can't really know anything just with a filename.